src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,181 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkTail(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size()) {
+            linkTail(value);
+        } else if (index == 0) {
+            linkHead(value);
+        } else {
+            insert(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            linkTail(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNode(index);
+        T oldItem = node.item;
+        node.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(findNode(index));
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        try {
+            unlink(findNode(object));
+            return true;
+        } catch (NoSuchElementException e) {
+            return false;
+        }
```

Comment:
We don't have to catch unchecked exceptions. Let's throw checked or you can just move the logic from that method here. Anyway you don't use it anywhere else

Reply:
moved logic from findNode by element in remove by element